lib/elastic_whenever/option.rb

```diff
@@ -133,10 +133,11 @@ def initialize(args)
                      elsif access_key && secret_key
                        Aws::Credentials.new(access_key, secret_key)
                      end
+      @credentials = nil unless @credentials&.set?
     end
 
     def aws_config
-      @aws_config ||= { credentials: credentials, region: region }.delete_if { |_k, v| v.nil? }
+      @aws_config ||= { credentials: credentials, region: region, profile: @credentials ? nil : profile }.delete_if { |_k, v| v.nil? }
```

Comment:
if the `@credentials` are nil due to whatever reason, we should pass in the profile in order to actually support the `--profile` option as it would otherwise just be ignored. A workaround could be to set the `AWS_PROFILE` env var in that case, but why pass in the `region`, too then?

Reply:
using SSO, it is not required to set credentials for an AWS SDK client at all, but you can switch profiles by passing in the profile as can be seen in this example:
```
irb(main):001> require 'aws-sdk-sts'
=> true
irb(main):002> sts_client = Aws::STS::Client.new
=> #<Aws::STS::Client>
irb(main):003> response = sts_client.get_caller_identity
=>
#<struct Aws::STS::Types::GetCallerIdentityResponse
...
irb(main):004> puts "Account: #{response.account}"
Account: 493373280XXX
=> nil
irb(main):005> sts_client = Aws::STS::Client.new profile: 'sofatutor-testing'
=> #<Aws::STS::Client>
irb(main):006> response = sts_client.get_caller_identity
=>
#<struct Aws::STS::Types::GetCallerIdentityResponse
...
irb(main):007> puts "Account: #{response.account}"
Account: 800587838XXX

Reply:
Thank you for your explanation. With this explanation, I understand that `profile` can be referenced other than shared credentials.